Updated readme and synopsis to reflect wording in P3411R2 (#23)
bemanproject/any_view: README.md

# beman.any_view: A generalized type-erased view with customizable properties

![Library Status](https://raw.githubusercontent.com/bemanproject/beman/refs/heads/main/images/badges/beman_badge-beman_library_under_development.svg)
![Continuous Integration Tests](https://github.com/bemanproject/any_view/actions/workflows/ci_tests.yml/badge.svg)
![Lint Check (pre-commit)](https://github.com/bemanproject/any_view/actions/workflows/pre-commit.yml/badge.svg)

**Implements**: `std::ranges::any_view` proposed in [any_view (P3411R2)](https://wg21.link/p3411r2).

**Status**: [Under development and not yet ready for production use.](https://github.com/bemanproject/beman/blob/main/docs/BEMAN_LIBRARY_MATURITY_MODEL.md#under-development-and-not-yet-ready-for-production-use)

> **Featured**: [C++Now 2025](https://schedule.cppnow.org/session/2025/a-view-for-any-occasion)

Reviewer: Maybe it's a better idea to link to the slides?

## Motivation

> `any_view` equips library APIs with the ability to declare function parameters and return types that are views, while

Reviewer: ```suggestion
`any_view` equips library APIs with the ability to declare function parameters and return types as views, while
```

Is this what you mean here?

    copyable      = 0b10000000,
};

constexpr any_view_options operator|(any_view_options, any_view_options) noexcept;
constexpr any_view_options operator&(any_view_options, any_view_options) noexcept;

template <class T>
struct /*rvalue-ref*/ {
    using type = T;
};

template <class T>
struct /*rvalue-ref*/<T&> {
    using type = T&&;
};

template <class T>
using /*rvalue-ref-t*/ = /*rvalue-ref*/<T>::type;

template <class RangeT, class AnyViewT>

> concept ext_any_compatible_viewable_range = /* see description */;

Author: I'll change it to `/* ... */` I guess? 🤷‍♂️ I don't really think the complete wording belongs in the readme, I need to add a `docs` directory and figure this out properly at some point. One thing I'm hoping this project will figure out soon is for documentation to get deployed to GitHub Pages from CI on every push to `main`. I've set that up plenty of times in my JavaScript repositories, but have not attempted it on any C++ repositories that I own.
